charts/kubernetes-agent/templates/nfs-pv.yaml

```diff
@@ -15,6 +15,9 @@ spec:
   mountOptions:
     - nfsvers=4.1
     - lookupcache=none
+    - soft
+    - bg
+    - timeo=5
```

Comment:
what do `soft` and `bg` do?

Reply:
I took a look at the man page to see what these did so I could accurately relay that info and realised that timeo was in tenths of a second. So I started experimenting again and found that:
1. `bg` is not required
2. there seems to be a slightly better setup than what I had originally (using `retrans` and `timeo`)
3. `soft` is required

Reply:
soft:
<img width="898" alt="Screenshot 2024-04-03 at 11 48 15" src="https://github.com/OctopusDeploy/helm-charts/assets/97430840/4c164dcf-dfbc-4f51-afb0-13a1a7203487">

Basically, if we don't have soft, it will never "fail" and will just hang forever trying to reconnect.

timeo:
<img width="887" alt="Screenshot 2024-04-03 at 11 48 23" src="https://github.com/OctopusDeploy/helm-charts/assets/97430840/7d19d6f3-5826-4eeb-83fc-4a1c741e91f1">

how long it waits before failing a connection in tenths of a second. The standard is 60 seconds.

retrans:
<img width="864" alt="Screenshot 2024-04-03 at 11 48 29" src="https://github.com/OctopusDeploy/helm-charts/assets/97430840/40a2782a-c111-4a72-a9dd-67bf3a118612">

how many times it will retry before failing.

With the settings I have now (`soft, timeo=5, retrans=5`) I get the following output in manual testing:
<img width="425" alt="Screenshot 2024-04-03 at 11 47 42" src="https://github.com/OctopusDeploy/helm-charts/assets/97430840/0956d544-46cc-4a9d-aa27-1c8cdccb8cfe">

I've done more tests where I saw it take up to 11.5 seconds to fail too so the range is roughly 3-12 seconds for accessing a broken mount to fail. If you restart the nfs pod, it returns instantly again.

Reply:
In terms of soft causing data corruption, I think if we say that "if file access fails once on a pod (tentacle or script pod) we will restart the tentacle deployment and fail the scripts in question" then we shouldn't have an issue with data corruption.